### PR TITLE
argcheck: restrict the type of elements in a list

### PR DESCRIPTION
**Pull request by njzjz**

### CODE REVIEW

**Comment by njzjz** [comment]
pre-commit.ci autofix

**Comment by njzjz** [comment]
I hope these changes can be carefully reviewed... I may make mistakes.

File: dpgen/data/arginfo.py [hunk 1]
```diff
@@ -105,25 +105,25 @@ def init_bulk_jdata_arginfo() -> Argument:
...
         [
-            Argument("stages", list, optional=False, doc=doc_stages),
-            Argument("elements", list, optional=False, doc=doc_elements),
-            Argument("potcars", list, optional=True, doc=doc_potcars),
+            Argument("stages", list[int], optional=False, doc=doc_stages),
+            Argument("elements", list[str], optional=False, doc=doc_elements),
+            Argument("potcars", list[str], optional=True, doc=doc_potcars),
             Argument("cell_type", str, optional=True, doc=doc_cell_type),
-            Argument("super_cell", list, optional=False, doc=doc_super_cell),
+            Argument("super_cell", list[int], optional=False, doc=doc_super_cell),
```
Comment (njzjz) [inline dpgen/data/arginfo.py:112]: Is it list[int] or list[float]?

File: dpgen/data/arginfo.py [hunk 2]
```diff
@@ -185,13 +185,13 @@ def init_surf_jdata_arginfo() -> Argument:
             Argument("z_min", int, optional=True, doc=doc_z_min),
             Argument("vacuum_max", float, optional=False, doc=doc_vacuum_max),
             Argument("vacuum_min", float, optional=True, doc=doc_vacuum_min),
-            Argument("vacuum_resol", list, optional=False, doc=doc_vacuum_resol),
+            Argument("vacuum_resol", list[float], optional=False, doc=doc_vacuum_resol),
             Argument("vacuum_numb", int, optional=True, doc=doc_vacuum_numb),
             Argument("mid_point", float, optional=True, doc=doc_mid_point),
             Argument("head_ratio", float, optional=True, doc=doc_head_ratio),
-            Argument("millers", list, optional=False, doc=doc_millers),
+            Argument("millers", list[list[int]], optional=False, doc=doc_millers),
```
Comment (njzjz) [inline dpgen/data/arginfo.py:192]: Is it list[list[int]] or list[list[float]]?

File: dpgen/generator/arginfo.py [hunk 3]
```diff
@@ -610,13 +621,13 @@ def fp_style_abacus_args() -> list[Argument]:
...
-        Argument("fp_pp_files", list, optional=False, doc=doc_fp_pp_files),
-        Argument("fp_orb_files", list, optional=True, doc=doc_fp_orb_files),
+        Argument("fp_pp_files", list[str], optional=False, doc=doc_fp_pp_files),
+        Argument("fp_orb_files", list[str], optional=True, doc=doc_fp_orb_files),
         Argument("fp_incar", str, optional=True, doc=doc_fp_incar),
         Argument("fp_kpt_file", str, optional=True, doc=doc_fp_kpt_file),
         Argument("fp_dpks_descriptor", str, optional=True, doc=doc_fp_dpks_descriptor),
         Argument("user_fp_params", dict, optional=True, doc=doc_user_fp_params),
-        Argument("k_points", list, optional=True, doc=doc_k_points),
+        Argument("k_points", list[int], optional=True, doc=doc_k_points),
```
Comment (njzjz) [inline dpgen/generator/arginfo.py:630]: Is it list[int] or list[float]?

File: dpgen/data/arginfo.py (hunk 1 above)
Comment (wanghan-iapcm) [reply dpgen/data/arginfo.py:112]: should be `int`. see https://pymatgen.org/pymatgen.core.html#pymatgen.core.structure.Structure.make_supercell

File: dpgen/data/arginfo.py (hunk 2 above)
Comment (wanghan-iapcm) [reply dpgen/data/arginfo.py:192]: The former.

File: dpgen/generator/arginfo.py (hunk 3 above)
Comment (wanghan-iapcm) [reply dpgen/generator/arginfo.py:630]: list[int]

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**